src/Worker/PluginExtractor.php

```diff
@@ -20,18 +20,32 @@ class PluginExtractor implements PluginExtractorInterface
     /** @var string */
     private $pluginFolder = '';
 
+    /** @var string */
+    private $legacyPluginFolder = '';
+
+    /** @var string[] Folders which will be used to decide if it is a legacy plugin */
+    private $legacyPluginRootFolders = [
+        'Backend',
+        'Core',
+        'Frontend',
+    ];
+
     /**
      * @param string $targetShopwareRoot root path to shopware installation where the plugin should be installed
      * @param string $pluginFolder       path to plugins inside the shop directory
```

Comment:
Do you want to add an annotation for the newly introduced parameter as well to highlight the differences between the parameters?